src/main/java/com/gaggle/contact/domain/Contact.java

```diff
@@ -45,12 +55,13 @@ public boolean equals(Object o) {
             return false;
         Contact contact = (Contact) o;
         return Objects.equals(this.id, contact.id) &&
-                Objects.equals(this.name, contact.name);
+                Objects.equals(this.name, contact.name) &&
+                Objects.equals(this.name, contact.email);
```

Comment:
looks like this.name should be this.email

Reply:
Possible null pointer exception, can we have a try catch here or some null checks?